h5io/_h5io.py

```diff
@@ -350,7 +350,28 @@ def _triage_write(
 
         if use_state:
             class_type = value.__class__.__module__ + "." + value.__class__.__name__
-            state = value.__getstate__()
+            reduced = value.__reduce__()
```

Comment:
Be careful with `__reduce__()` that does not work for `pathlib` - that is why `pathlib` is part of the tests.

Reply:
I've added a small workaround to not raise an error if the function returned by reduce is the class constructor itself as is the case for pathlib objects.